src/graph/bipartite_matching.rs

```diff
@@ -15,15 +17,15 @@ impl BipartiteMatching {
             adj: vec![vec![]; num_vertices_grp1 + 1],
             num_vertices_grp1,
             num_vertices_grp2,
-            mt: vec![-1; num_vertices_grp2 + 1],
+            mt2: vec![-1; num_vertices_grp2 + 1],
+            mt1: vec![-1; num_vertices_grp1 + 1],
             used: vec![false; num_vertices_grp1 + 1],
         }
     }
     #[inline]
-    // Add an undirected edge u-v in the graph
+    // Add an directed edge u->v in the graph
     pub fn add_edge(&mut self, u: usize, v: usize) {
         self.adj[u].push(v);
-        // self.adj[v].push(u);
```

Comment:
Why did you change it to be directed?

Reply:
Because we are only pushing v into adj[u].
If it were undirected we would have
adj[u].push(v) and adj[v].push(u)